packages/utils/typescript/lib/generators/schemas/attributes.js

```diff
@@ -0,0 +1,241 @@
+    },
+    media() {
+      return ['MediaAttribute'];
+    },
+    relation() {
+      const { relation, target } = attribute;
+
+      if (relation.includes('morph') | relation.includes('Morph')) {
+        return [
+          'RelationAttribute',
+          [factory.createStringLiteral(uid, true), factory.createStringLiteral(relation, true)],
+        ];
+      }
+
+      return [
+        'RelationAttribute',
+        [
+          factory.createStringLiteral(uid, true),
+          factory.createStringLiteral(relation, true),
+          factory.createStringLiteral(target, true),
+        ],
+      ];
+    },
+    component() {
+      const target = attribute.component;
+      const params = [factory.createStringLiteral(target, true)];
+
+      if (attribute.repeatable) {
+        params.push(factory.createTrue());
+      }
+
+      return ['ComponentAttribute', params];
+    },
+    dynamiczone() {
+      const componentsParam = factory.createTupleTypeNode(
+        attribute.components.map(component => factory.createStringLiteral(component))
+      );
+
+      return ['DynamicZoneAttribute', [componentsParam]];
+    },
+  };
+
+  if (!Object.keys(mappers).includes(attribute.type)) {
+    console.warning(
+      `"${attributeName}" attribute from "${uid}" has an invalid type: "${attribute.type}"`
+    );
+
+    return null;
+  }
+
+  const [attributeType, typeParams] = mappers[attribute.type]();
+
+  addImport(attributeType);
+
+  return getTypeNode(attributeType, typeParams);
+};
+
+/**
+ * Collect every modifier node from an attribute
+ *
+ * @param {string} _attributeName
+ * @param {object} attribute
+ * @returns {object[]}
+ */
+const getAttributeModifiers = (_attributeName, attribute) => {
+  const modifiers = [];
+
+  // Required
+  if (attribute.required) {
+    addImport('RequiredAttribute');
+
+    modifiers.push(factory.createTypeReferenceNode(factory.createIdentifier('RequiredAttribute')));
+  }
+
+  // Private
+  if (attribute.private) {
+    addImport('PrivateAttribute');
+
+    modifiers.push(factory.createTypeReferenceNode(factory.createIdentifier('PrivateAttribute')));
+  }
+
+  // Unique
+  if (attribute.unique) {
+    addImport('UniqueAttribute');
+
+    modifiers.push(factory.createTypeReferenceNode(factory.createIdentifier('UniqueAttribute')));
+  }
+
+  // Configurable
+  if (attribute.configurable) {
+    addImport('ConfigurableAttribute');
+
+    modifiers.push(
+      factory.createTypeReferenceNode(factory.createIdentifier('ConfigurableAttribute'))
+    );
+  }
+
+  // Plugin Options
+  if (!fp.isEmpty(attribute.pluginOptions)) {
+    addImport('SetPluginOptions');
+
+    modifiers.push(
+      factory.createTypeReferenceNode(
+        factory.createIdentifier('SetPluginOptions'),
+        // Transform the pluginOptions object into an object litteral expression
+        [toTypeLitteral(attribute.pluginOptions)]
+      )
+    );
+  }
+
+  // Min / Max
+  if (!fp.isNil(attribute.min) || !fp.isNil(attribute.max)) {
```

Comment:
shouldn't this be managed by attribute type instead? 
Looks like it's missing other modifiers (UID one / regex ) ?